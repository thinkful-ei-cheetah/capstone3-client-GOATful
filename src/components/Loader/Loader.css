.loader {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  position: fixed;
  background-color: rgba(0,0,0,0.8);
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 10;
  font-size: 20px;
  color: white;
}

.sk-cube-grid {
  width: 50px !important;
  height: 50px !important;
}