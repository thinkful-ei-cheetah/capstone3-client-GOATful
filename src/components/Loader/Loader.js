import React from 'react';
import './Loader.css';
import Spinner from 'react-spinkit'
  
export default function Loader(props) {
  if (props.isLoading) {
    return (
      <div className='loader'>
        <Spinner name="cube-grid" color='white' fadeIn='quarter'/>
        <p>loading</p>
      </div>
    )
  } else {
    return <></>
  }
  
}
