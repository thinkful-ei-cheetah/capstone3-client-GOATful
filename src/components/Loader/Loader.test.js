import React from 'react';
import ReactDOM from 'react-dom';
import Loader from './Loader';
import renderer from 'react-test-renderer';


describe('Loader Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< Loader />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Loader/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
