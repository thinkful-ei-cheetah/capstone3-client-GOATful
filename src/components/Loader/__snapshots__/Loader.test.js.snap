// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders the UI as expected 1`] = `null`;
